arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -360,11 +357,11 @@ Z_INTERNAL void crc_fold_copy(unsigned int crc0[4 * 5], unsigned char *dst, cons
     partial_fold((size_t)len, &xmm_crc0, &xmm_crc1, &xmm_crc2, &xmm_crc3, &xmm_crc_part);
 done:
     /* CRC_SAVE */
-    _mm_storeu_si128((__m128i *)crc0 + 0, xmm_crc0);
-    _mm_storeu_si128((__m128i *)crc0 + 1, xmm_crc1);
-    _mm_storeu_si128((__m128i *)crc0 + 2, xmm_crc2);
-    _mm_storeu_si128((__m128i *)crc0 + 3, xmm_crc3);
-    _mm_storeu_si128((__m128i *)crc0 + 4, xmm_crc_part);
+    _mm_storeu_si128((__m128i *)crc->fold + 0, xmm_crc0);
+    _mm_storeu_si128((__m128i *)crc->fold + 1, xmm_crc1);
+    _mm_storeu_si128((__m128i *)crc->fold + 2, xmm_crc2);
+    _mm_storeu_si128((__m128i *)crc->fold + 3, xmm_crc3);
+    _mm_storeu_si128((__m128i *)crc->fold + 4, xmm_crc_part);
```

Comment:
Using temporary variable would make this cleaner as it would remove need for repeated casting.
Using unaligned version should also be unnecessary as the structure is guaranteed to be aligned and fold is the first member of the structure.

Reply:
Fixed.